tests/proxytest.py

```diff
@@ -101,95 +101,185 @@ async def wait_for_services(urls):
+            except KeyError:
+                pass
+
+            assert route == expected_output(spec, backend.geturl())
+
+            # Test the actual routing
+            responding_backend1 = await utils.get_responding_backend_port(
+                proxy_url, normalize_spec(spec)
+            )
+            responding_backend2 = await utils.get_responding_backend_port(
+                proxy_url, normalize_spec(spec) + "something"
+            )
+            assert (
+                responding_backend1 == backend.port
+                and responding_backend2 == backend.port
+            )
+
+    for i, spec in enumerate(existing_routes, start=1):
+        backend = default_backend._replace(
+            netloc=f"{default_backend.hostname}:{default_backend.port+i}"
+        )
+        launch_backend(backend.port, backend.scheme)
+        extra_backends.append(backend)
+
+    launch_backend(default_backend.port, default_backend.scheme)
+    await wait_for_services(
+        [proxy.public_url, default_backend.geturl()]
+        + [backend.geturl() for backend in extra_backends]
+    )
+
+    """ Create existing routes """
+    for i, spec in enumerate(existing_routes):
         try:
-            del route["data"]["last_activity"]  # CHP
-        except KeyError:
+            await proxy.add_route(spec, extra_backends[i].geturl(), copy.copy(data))
+        except Exception:
             pass
-        finally:
-            assert route == expected_output
-        if proxy.public_url.endswith("/"):
-            req_url = proxy.public_url[:-1]
-        else:
-            req_url = proxy.public_url
-        # Test the actual routing
-        responding_backend1 = await utils.get_responding_backend_port(
-            req_url, routespec
-        )
-        responding_backend2 = await utils.get_responding_backend_port(
-            req_url, routespec + "/something"
-        )
-        assert (
-            responding_backend1 == backend_port and responding_backend2 == backend_port
-        )
 
-    """ Test delete + get """
-    with context():
+    def finalizer():
+        async def cleanup():
+            """ Cleanup """
+            for spec in existing_routes:
+                try:
+                    await proxy.delete_route(spec)
+                except Exception:
+                    pass
+
+        event_loop.run_until_complete(cleanup())
+
+    request.addfinalizer(finalizer)
+
+    """ Test add """
+    with context(routespec):
+        await proxy.add_route(routespec, default_backend.geturl(), copy.copy(data))
+
+    """ Test get """
+    await test_route_exist(routespec, default_backend)
+    for i, spec in enumerate(existing_routes):
+        await test_route_exist(spec, extra_backends[i])
+
+    """ Test delete """
+    with context(routespec):
         await proxy.delete_route(routespec)
         route = await proxy.get_route(routespec)
-    if not expect_value_error:
+
+    """ Test get """
```

Comment:
The next part tests that the `delete` operation had the desired outcome, so maybe it makes sense to delete this comment. What do you think?

Reply:
I've added more details to the comment